buildpack.md

```diff
@@ -744,32 +750,26 @@ The stack ID:
 
 The stack `build-images` and `run-images` are suggested sources of images for platforms that are unaware of the stack ID. Buildpack authors MUST ensure that these images include all mixins specified in `mixins`.
 
-### launch.toml (TOML)
+### app.toml (TOML)
 
 ```toml
 [[processes]]
 type = "<process type>"
 command = "<command>"
-```
-
-Buildpacks MUST specify:
-
-- A unique process type for each entry within a `launch.toml` file.
-- A command that is valid when executed using the Bash 3+ shell.
-
-### develop.toml (TOML)
 
-```toml
-[[processes]]
-type = "<process type>"
-command = "<command>"
+[[slices]]
+path = "<app sub-path>"
 ```
 
-Buildpacks MUST specify:
+The buildpack MAY specify any number of processes or slices.
+
+For each process, the buildpack MUST specify:
 
-- A unique process type for each entry within a `develop.toml` file.
+- A unique process type for each entry within a `app.toml` file.
 - A command that is valid when executed using the Bash 3+ shell.
 
+For each slice, buildpacks MUST specify a path relative to the root of the app directory.
```

Comment:
What's the reasoning for relative paths here?  This is the only location in the spec (I think) that relative paths are used.  From an implementation point of view, this is just a bit more trouble (calculating relative paths) and given that relative paths could be parent relative (e.g. `../../bin/rm`) it seems that the lifecycle would still be responsible for doing directory-traversal enforcement.

Reply:
For portability across stacks that may define the app directory differently, I don't think these paths should be absolute to the filesystem root (e.g., `/workspace/...`).

I think my preference would be to tolerate:
1. Absolute paths, with an assumption that the app dir is the root (`/`).
2. Relative paths from the app dir, ignoring all paths that aren't rooted at the app dir.

Directly traversal enforcement is necessary in either case. A false root directory can be just as susceptible to malicious directory traversal as a relative path.